algorithms/kernel/k_nearest_neighbors/oneapi/bf_knn_classification_predict_kernel_ucapi_impl.i

```diff
@@ -144,145 +114,119 @@ Status KNNClassificationPredictKernelUCAPI<algorithmFpType>::
-            auto classesRows = classes.template get<int>().toHost(ReadWriteMode::readOnly);
-            auto res = resultBlock.getBuffer().toHost(ReadWriteMode::writeOnly);
-                for(int i = 0; i < curProbeBlockSize; i++)
-                {
-                    res.get()[pfirst + i] = classesRows.get()[i];
-                }
+            auto classesRows = classes.template get<int>().toHost(ReadWriteMode::readOnly, &st);
+            DAAL_CHECK_STATUS_VAR(st);
+            auto res = resultBlock.getBuffer().toHost(ReadWriteMode::writeOnly, &st);
+            DAAL_CHECK_STATUS_VAR(st);
+            for (int i = 0; i < curProbeBlockSize; i++) res.get()[pfirst + i] = classesRows.get()[i];
             DAAL_CHECK_STATUS_VAR(y->releaseBlockOfRows(resultBlock));
         }
+        DAAL_CHECK_STATUS_VAR(ntData->releaseBlockOfRows(probeRows));
     }
-    DAAL_CHECK_STATUS_VAR(ntData->releaseBlockOfRows(probeRows));
-    DAAL_CHECK_STATUS_VAR(points->releaseBlockOfRows(dataRows));
-    DAAL_CHECK_STATUS_VAR(labels->releaseBlockOfRows(labelRows));
     return st;
 }
-
-template <typename algorithmFpType>
-void KNNClassificationPredictKernelUCAPI<algorithmFpType>::initCategories
-        (ExecutionContextIface& context,
-         const KernelPtr& kernel_init_categories,
-         const Buffer<int>& labels,
-         UniversalBuffer& categories,
-         uint32_t probeBlockSize,
-         uint32_t dataBlockSize,
-         uint32_t offset,
-         Status* st)
-{
-    DAAL_ITTNOTIFY_SCOPED_TASK(compute.init_categories);
-    KernelArguments args(4);
-    args.set(0, labels, AccessModeIds::read);
-    args.set(1, categories, AccessModeIds::write);
-    args.set(2, dataBlockSize);
-    args.set(3, offset);
-
-    KernelRange global_range(dataBlockSize, probeBlockSize);
-
-    {
-        context.run(global_range, kernel_init_categories, args, st);
-    }
-}
-
 template <typename algorithmFpType>
-void KNNClassificationPredictKernelUCAPI<algorithmFpType>::copyPartialSelections
-        (ExecutionContextIface& context,
-         const KernelPtr& kernel_gather_selection,
-         UniversalBuffer& distances,
-         UniversalBuffer& categories,
-         UniversalBuffer& partialDistances,
-         UniversalBuffer& partialCategories,
-         uint32_t probeBlockSize,
-         uint32_t nK,
-         uint32_t& nPart,
-         uint32_t totalParts,
-         bool bKeepPrevious,
-         Status* st)
+void KNNClassificationPredictKernelUCAPI<algorithmFpType>::copyPartialSelections(
+    ExecutionContextIface & context, const KernelPtr & kernel_gather_selection, UniversalBuffer & distances, UniversalBuffer & categories,
+    UniversalBuffer & partialDistances, UniversalBuffer & partialCategories, uint32_t probeBlockSize, uint32_t nK, uint32_t & nPart,
+    uint32_t totalParts, bool bKeepPrevious, Status * st)
 {
     DAAL_ITTNOTIFY_SCOPED_TASK(compute.copyPartialSelections);
-    if(bKeepPrevious)
+    if (bKeepPrevious)
```

Comment:
I suggest to move such logic outside of this function. Is it the responsibility of `copyPartialSelections` method to update `nPart`?